### DISCUSSION TITLE
Paopao网关，用虚拟机生成ios

### DISCUSSION THREAD

**Discussion by Mxzjxs** [kkkgo/PaoPaoGateWay#93] [问和答]
昨天搞了几个小时终于搭建好了，目前使用体验是最好的，感觉比dae都更好用，不过就是有几个问题，第一个用虚拟机编译ios的方式安装，我想更改节点信息有哪些方式，我每次都是重新编译。第二个就是节点规则有详细介绍使用方法吗，就比如我奈飞用不同节点如何实现呢，把规则信息统一放进去编译？第三个就是ui和内核啥时候可以自己替换呢，比如换singbox内核或者新ui。最后就是想问是否有telegram群组啊😃

**Comment by kkkgo** [comment]
1、如果你是用docker定制ISO的话，所有文件都是**按需内嵌**的，custom.yaml、ppgw.ini等都是非必须内嵌的文件，没有被内嵌的文件会按照默认的DHCP方式配置。在yaml模式下，会自动使用拉取ppgw.ini的方式拉取配置，你可以把你的配置文件放在指定的http服务，详情参考文档”使用DHCP下发配置“一节。如果在suburl模式下，则会从指定的url下载yaml配置。
2、节点规则请直接参考clash/mihomo官方文档，或者可以参考subconverter项目。
3、ui是功能定制和性能优化的，不可更换。你喜欢的话可以自己搭一个web静态放你喜欢的ui填api地址。更换内核请参考docker定制一节，可以自由更换clash内核，如果当前内核已经满足需求不建议更换，只会浪费内存，除非协议不支持等。singbox并不兼容clash配置以及命令调用，无法更换，而且支持的协议和分流规则并无差异，如果不知道自己的需求，换了也没啥意义。

**Reply by LLIce** [reply]
看到这才理解，原来是得打包的时候不放custom.yaml和ppgw.ini文件。我说怎么开了http服务，把custom.yaml放data目录下怎么修改都没变化，每次调试配置文件都得打包，打了好几十个出来。

**Reply by Mxzjxs** [reply]
> 都

哈哈哈，我之前也是这样。话说这个http服务在哪儿开启

**Reply by kkkgo** [reply]
你可以自建http服务或者查阅paopaodns文档的HTTP_FILE选项。

**Reply by Mxzjxs** [reply]
> 你可以自建http服务或者查阅paopaodns文档的HTTP_FILE选项。

"如果配合PaoPaoDNS使用，你只需要设置SERVER_IP参数和设置HTTP_FILE=yes，映射7889端口即可，你可以直接把配置文件ppgw.ini放在PaoPaoDNS的/data目录。"

这段话的意思是，我需要在ppgw.ini里面输入httpfile =yes ，serverip则是输入paopaodns的局域网地址加7889端口，就可以把文件放入data里面就可以了吗？

**Reply by kkkgo** [reply]
不是的。`SERVER_IP`和`HTTP_FILE`是PaoPaoDNS的环境变量参数，你应该看PaoPaoDNS的文档。

**Reply by Mxzjxs** [reply]
> 不是的。`SERVER_IP`和`HTTP_FILE`是PaoPaoDNS的环境变量参数，你应该看PaoPaoDNS的文档。

我明白了，刚设置了下，我想问下就是paopaodns改好后，必须把paopaogateway的dns地址改成paopaodns的地址，才能正常拉取。那么paopaogateway的dns的作用就只是拉取文件吗，会影响科学速度吗。第二个 那个dns burn我看默认是开启的，是不是搭配paopaodns必须开启这个，然后填paopaodns的地址。

**Reply by kkkgo** [reply]
1、是的，网卡的DNS信息仅用于拉取配置文件，与代理速度无关。其他行为使用ppgw.ini的DNS设置。
```yaml
# Set your trusted DNS here
# default: dns_ip=1.0.0.1
dns_ip=10.10.10.8
# default: dns_port=53
# If used with PaoPaoDNS, you can set the 5304 port
dns_port=5304
```
2、没有默认开启的功能，dns_burn选项依赖fast_node选项，只有fast_node设置为yes的时候dns_burn选项设置为yes才会生效，作用请参考文档。

**Reply by Mxzjxs** [reply]
> 1、是的，网卡的DNS信息仅用于拉取配置文件，与代理速度无关。其他行为使用ppgw.ini的DNS设置。
> 
> ```yaml
> 
> # Set your trusted DNS here
> 
> # default: dns_ip=1.0.0.1
> 
> dns_ip=10.10.10.8
> 
> # default: dns_port=53
> 
> # If used with PaoPaoDNS, you can set the 5304 port
> 
> dns_port=5304
> 
> ```
> 
> 2、没有默认开启的功能，dns_burn选项依赖fast_node选项，只有fast_node设置为yes的时候dns_burn选项设置为yes才会生效，作用请参考文档。

这个问题我懂了，完整检测报错是啥情况![image](https://github.com/kkkgo/PaoPaoGateWay/assets/168628534/ffc5b31c-9413-460b-88db-c869fc74f22a)

**Reply by kkkgo** [reply]
你可以跑debug.sh去paopaodns讨论区提问或者搜索答案。
